test/suites/upgrade/downgrade-block.robot

```diff
@@ -26,7 +26,17 @@ Downgrade Is Blocked
 
     ${initial_deploy_backup}=    Get Future Backup Name For Current Boot
 
-    Deploy Commit Expecting A Rollback    ${OLDER_MICROSHIFT_REF}    write_agent_cfg=False
+    ${is_bootc}=    Is System Bootc
+    IF    ${is_bootc}
+        Deploy Bootc Commit Expecting A Rollback
+        ...    ${BOOTC_REGISTRY}
+        ...    ${OLDER_MICROSHIFT_REF}
+        ...    write_agent_cfg=False
+    ELSE
+        Deploy Commit Expecting A Rollback
+        ...    ${OLDER_MICROSHIFT_REF}
+        ...    write_agent_cfg=False
+    END
```

Comment:
I'm wondering if we could move bootc-or-ostree` logic to different place? Like "Deploy Commit Expecting A Rollback" could check if system is bootc and we could pass the bootc url+ref as single string in the scenario script instead of joining them later.
If it's about allowing for insecure registry, we could use optional arguments like:
"if registry_url is given, it's a bootc, allow insecure, then concat url+ref".

Because doing this in every upgrade test individually scales poorly imho

Reply:
Let me try this.